### DISCUSSION TITLE
Is there any way to setup NetPad to use our DbContext defined in our project dll, like LINQPad Entity Framework Core Driver

### DISCUSSION THREAD

**Discussion by jwallet** [tareqimbasher/NetPad#72] [Q&A] [answered]
Hi,

I'm wondering if I can use my project dll and return the dbContext from there so I can query stuff using my entities, example I have a table TimeSheets but in my dll I made two variants NonBillableTimeSheet and BillableTimeSheet models that are based on a discriminator, I can seem to only use the database collections in NetPad which does not provide those types.

LINQPad driver in example: 
![image](https://github.com/tareqimbasher/NetPad/assets/23088305/2f7e8c6e-72ff-4328-b01e-46ba6f5ecfc3)

**Comment by jwallet** [comment, marked as answer]
Okay, I actually found out the solution:
> `.UseNpgsql` to work with my PostGreSQL database

```c#
var optionsBuilder = new DbContextOptionsBuilder<AppDbContext>();
optionsBuilder.UseNpgsql("YOUR_CONNECTION_STRING", x =>
  {
    x.EnableRetryOnFailure();
  })
  .LogTo(output =>
  {
      // Log to SQL tab
      DumpExtension.Sink.SqlWrite(output + "\n"); // >= v0.7.2
      // ScriptRuntimeServices.SqlWrite(output + "\n"); // < v0.7.2
  });
var dbContextOptions = optionsBuilder.Options;

// specify all injected depedencies, mine was DbContextOptions and an optional Mediator
var _appDbContext = new AppDbContext(dbContextOptions, null);
```

**Comment by tareqimbasher** [comment]
Currently NetPad only offers the option to build the data context automatically when adding a database connection. The ability to select an assembly that already has a data context and have NetPad use that assembly instead of its auto-generated one is not yet available, although its definitely something planned.

One option is to add and reference your assembly in script properties (F4 and add your assembly), and then get your `DbContext` and pass it the connection string, DbContextOptions...etc. You wouldn't need to add a data connection from within NetPad itself, you just reference your assembly and use it. I haven't tried this however, YMMV.
